File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def days(date):
    y, m, d = map(int, date.split("."))
    return y * 28 * 12 + m * 28 + d

def solution(today, terms, privacies):
    months = {v[0]: int(v[2:]) * 28 for v in terms}
    today = days(today)
    expire = [
        i + 1 for i, privacy in enumerate(privacies)
        if days(privacy[:-2]) + months[privacy[-1]] <= today
    ]
    return expire

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
  let answer = [];
  let [y, m, d] = today.split(".").map(Number);
  const tdates = y * 12 * 28 + m * 28 + d;
  let term = {};

  terms.map(v => {
    let [type, n] = v.split(" ");
    term[type] = +n;
  });
    
  privacies.map((v, i) => {
    let [date, type] = v.split(" ");
    date = date.split(".").map(Number);
    let dates = date[0] * 12 * 28 + date[1] * 28 + date[2] + term[type] * 28;
    if (dates <= tdates) answer.push(i + 1);
  });
  return answer;
}